url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/org/jbox2d/dynamics/contacts/ContactSolver.java

```diff
@@ -0,0 +1,908 @@
+				
+//				temp.set(normal).mulLocal(ccp.normalImpulse);
+//				P.set(tangent).mulLocal(ccp.tangentImpulse).addLocal(temp);
+//				bodyA.m_angularVelocity -= invIA * Vec2.cross(ccp.rA, P);
+//				temp.set(P).mulLocal(invMassA);
+//				bodyA.m_linearVelocity.subLocal(temp);
+//				bodyB.m_angularVelocity += invIB * Vec2.cross(ccp.rB, P);
+//				temp.set(P).mulLocal(invMassB);
+//				bodyB.m_linearVelocity.addLocal(temp);
+				
+				
+				final float Px = ccp.normalImpulse * normal.x + ccp.tangentImpulse * tangent.x;
+				final float Py = ccp.normalImpulse * normal.y + ccp.tangentImpulse * tangent.y;
+
+				bodyA.m_angularVelocity -= invIA * (ccp.rA.x * Py - ccp.rA.y * Px);
+				bodyA.m_linearVelocity.x -= Px * invMassA;
+				bodyA.m_linearVelocity.y -= Py * invMassA;
+				
+				bodyB.m_angularVelocity += invIB * (ccp.rB.x * Py - ccp.rB.y * Px);
+				bodyB.m_linearVelocity.x += Px * invMassB;
+				bodyB.m_linearVelocity.y += Py * invMassB;
+			}
+		}
+	}
+	
+	// djm pooling, and from above
+	
+//	public final void initVelocityConstraints(TimeStep step){
+//		// Warm start.
+//		for (int i = 0; i < m_constraintCount; ++i){
+//			ContactConstraint c = m_constraints[i];
+//
+//			Body bodyA = c.bodyA;
+//			Body bodyB = c.bodyB;
+//			float invMassA = bodyA.m_invMass;
+//			float invIA = bodyA.m_invI;
+//			float invMassB = bodyB.m_invMass;
+//			float invIB = bodyB.m_invI;
+//			Vec2 normal = c.normal;
+//			Vec2.crossToOut(normal, 1.0f, tangent);
+//
+//			if (step.warmStarting){
+//				for (int j = 0; j < c.pointCount; ++j){
+//					ContactConstraintPoint ccp = c.points[j];
+//					ccp.normalImpulse *= step.dtRatio;
+//					ccp.tangentImpulse *= step.dtRatio;
+//					//Vec2 P = ccp.normalImpulse * normal + ccp.tangentImpulse * tangent;
+//					temp1.set(normal).mulLocal(ccp.normalImpulse);
+//					P.set(tangent).mulLocal(ccp.tangentImpulse).addLocal(temp1);
+//
+//					bodyA.m_angularVelocity -= invIA * Vec2.cross(ccp.rA, P);
+//					//bodyA.m_linearVelocity -= invMassA * P;
+//					temp1.set(P).mulLocal(invMassA);
+//					bodyA.m_linearVelocity.subLocal(temp1);
+//
+//					bodyB.m_angularVelocity += invIB * Vec2.cross(ccp.rB, P);
+//					//bodyB.m_linearVelocity += invMassB * P;
+//					temp1.set(P).mulLocal(invMassB);
+//					bodyB.m_linearVelocity.addLocal(temp1);
+//				}
+//			}
+//			else{
+//				for (int j = 0; j < c.pointCount; ++j){
+//					ContactConstraintPoint ccp = c.points[j];
+//					ccp.normalImpulse = 0.0f;
+//					ccp.tangentImpulse = 0.0f;
+//				}
+//			}
+//		}
+//	}
+	
+	//djm pooling from above
+	private final Vec2 dv = new Vec2();
+	private final Vec2 a = new Vec2();
+	private final Vec2 b = new Vec2();
+	private final Vec2 dv1 = new Vec2();
+	private final Vec2 dv2 = new Vec2();
+	private final Vec2 x = new Vec2();
+	private final Vec2 d = new Vec2();
+	private final Vec2 P1 = new Vec2();
+	private final Vec2 P2 = new Vec2();
+	
+	public final void solveVelocityConstraints(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 144 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.